common/browser_context.go

```diff
@@ -439,14 +439,58 @@ func (b *BrowserContext) runWaitForEventHandler(
 				return
 			}
 
-			if retVal, err := predicateFn(b.vu.Runtime().ToValue(p)); err == nil && retVal.ToBoolean() {
+			// TODO: Find a better place to instantiate a page/context/iteration
+			//       scoped taskqueue.
+			p.createTaskQueue()
+
+			tempPassed := make(chan bool)
+			tempErrOut := make(chan error)
+			// TODO: Is this is a suitable place to add the predicate function to
```

Comment:
Is it ok to add things to the taskqueue within the business logic or should we try to add it in the mapping layer?

Reply:
Blending `TaskQueue` into the business logic binds us more to Goja and k6 instead of native Go channel constructs. This makes tests complicated and codes error-prone if we continue to let it infiltrate more. The [tests look more like event loop event handling](https://github.com/grafana/xk6-browser/blob/main/tests/page_test.go#L923-L983) instead of focusing on the actual logic. This makes them harder to craft and understand and makes them error-prone. 

I have a strong opinion but can hold it weakly, depending on our discussions :) It's fine to continue blending it to make progress until we find a better way, though 👍